### DISCUSSION TITLE
Is anyone interested in a toolkit for Lexical?

### DISCUSSION THREAD

**Discussion by RegiByte** [facebook/lexical#4541] [Ideas]
I've read through most of Lexical's source code and I gotta say it's very impressive.

I've built countless editors with ProseMirror and I can speak about the pain that it is to find good plugins and to create your own, the ProseMirror API is confusing and not easy to grasp for newcomers.

That being said, I'd like to share with the Lexical community a small library that I'm working on, It's called Lexi-kit (I hope this doesn't infringe any restrictions because of the name).

Essentially is a collection of headless plug-n-play plugins and nodes.

Some of them are directly inspired by the plugins we see on [The playground](https://playground.lexical.dev/). I have seen in a couple of discussions here that the Core team has no time (nor should they) to turn all those plugins into separate packages so that's what I'm doing, slowly but surely.

I plan to add more plugins in the upcoming weeks/months and I'd like to get some feedback from users regarding the API style and plugin ideas.

Any feedback is welcome, all packages are exported as esm and cjs modules.

https://github.com/RegiByte/lexi-kit

**Comment by moy2010** [comment]
Sounds like a good idea! I would love to contribute if I have some spare time for it.

**Reply by RegiByte** [reply]
I'm glad to hear! Any help you need just let me know, I'll be working on the project in my spare time too.

**Comment by richardaum** [comment]
This is a good initiative. You could get more visibility on this if the official team add a link to it on their docs/readme.
